File: sdk/go/command_approval_test.go
package colossus

import (
	v1alpha1 "github.com/obscuritylabs/colossus/sdk/go/gen/colossus/api/v1alpha1"
	"google.golang.org/protobuf/proto"
	"testing"
)

func TestCommandApprovalOptionalContextRoundTrip(t *testing.T) {
	approval := &v1alpha1.ApprovalInteraction{Reason: "Approval required", Action: "process.execute", Resource: "configured executable", RequestHash: "binding"}
	for _, context := range []*v1alpha1.CommandApprovalContext{nil, {
		Justification: "Check dependency versions.", Executable: "/bin/sh",
		Arguments: []string{"-c", "echo 'two  spaces'", "", "é"}, WorkingDirectory: "/work/project", Redacted: true,
	}} {
		approval.CommandContext = context
		encoded, err := proto.Marshal(approval)
		if err != nil {
			t.Fatal(err)
		}
		var decoded v1alpha1.ApprovalInteraction
		if err := proto.Unmarshal(encoded, &decoded); err != nil {
			t.Fatal(err)
		}
		if !proto.Equal(approval, &decoded) {
			t.Fatal("command context changed during round trip")
		}
	}
	if approval.ProtoReflect().Descriptor().Fields().ByName("command_context").Number() != 8 {
		t.Fatal("command context field number changed")
	}
}

File: sdk/typescript/test/command-approval.test.ts
import assert from "node:assert/strict";
import { test } from "node:test";
import { ApprovalInteraction } from "../src/gen/colossus/api/v1alpha1/agent_run.js";

test("command context round trips without collapsing argv and legacy absence remains absent", () => {
  const legacy = ApprovalInteraction.fromPartial({
    reason: "Approval required",
    action: "process.execute",
    resource: "configured executable",
    requestHash: "binding",
  });
  assert.equal(
    ApprovalInteraction.decode(ApprovalInteraction.encode(legacy).finish())
      .commandContext,
    undefined,
  );
  const approval = ApprovalInteraction.fromPartial({
    ...legacy,
    commandContext: {
      justification: "Check dependency versions.",
      executable: "/bin/sh",
      arguments: ["-c", "echo 'two  spaces'", "", "é"],
      workingDirectory: "/work/project",
      redacted: true,
    },
  });
  assert.deepEqual(
    ApprovalInteraction.decode(ApprovalInteraction.encode(approval).finish()),
    approval,
  );
});
